### DISCUSSION TITLE
Creating a did:web

### DISCUSSION THREAD

**Discussion by MajdT51** [walt-id/.github#48] [Help] [answered]
Hello, I have a question:

I have the SSI Kit running in docker on my laptop. In the Custody API i would like to create a web did.
from the docs I see:

curl -X 'POST' \
  'https://custodian.ssikit.walt.id/did/create' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "method": "web",
  "didWebDomain": "walt.id"
}'

I'm a little bit confused. Is "walt.id" a placeholder that should be replaced (with what) ? where is the did document stored and will a new key be generated for it or do i need to assign it to one of the (imported or generated)  keyIds?

Thanks

**Comment by severinstampler** [comment, marked as answer]
Hi
When you create a did:web, it is usually in your own responsibility to host the DID document on your domain.
The DID document must be hosted on the specified domain on a well-known path:
e.g.: for the DID
`did:web:example.com`
The DID document must be hosted at:
`https://example.com/.well-known/did.json`

It is also possible to specify sub-paths for the did document, which must be included in the did:
e.g., for the DID
`did:web:w3c-ccg.github.io:user:alice`
the DID document must be hosted at:
`https://w3c-ccg.github.io/user/alice/did.json`

See also the instructions in the "Create (Register)" section of the did:web specification:
https://w3c-ccg.github.io/did-method-web/#create-register 

Here is an example of DID creation requests and responses, using the **SSI Kit Custodian REST API**:

**1. Generate key:**

_Request_:

```
curl -X 'POST' \
  'https://custodian.ssikit.walt.id/keys/generate' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{ "keyAlgorithm": "EdDSA_Ed25519" }'
```

_Response_:

```
{
  "algorithm": "EdDSA_Ed25519",
  "cryptoProvider": "SUN",
  "keyId": {
    "id": "348ff0f0d0914251a625dfeac31a7470"
  },
  "keyPair": {},
  "keysetHandle": null
}
```

**2. Generate DID**

_Request_:

```
curl -X 'POST' \
  'https://custodian.ssikit.walt.id/did/create' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{ "method": "web", "keyAlias": "348ff0f0d0914251a625dfeac31a7470", "domain": "example.com", "path": "user/alice" 
}'
```

_Response_:

```
did:web:example.com:user:alice
```

**3. Fetch DID document**

_Request_:

```
curl -X 'GET' \
  'https://custodian.ssikit.walt.id/did/did%3Aweb%3Aexample.com%3Auser%3Aalice' \
  -H 'accept: application/json'
```

_Response_:

```
{
  "assertionMethod": [
    {
      "controller": "",
      "id": "did:web:example.com:user:alice#348ff0f0d0914251a625dfeac31a7470",
      "type": ""
    }
  ],
  "authentication": [
    {
      "controller": "",
      "id": "did:web:example.com:user:alice#348ff0f0d0914251a625dfeac31a7470",
      "type": ""
    }
  ],
  "@context": [
    "https://www.w3.org/ns/did/v1"
  ],
  "id": "did:web:example.com:user:alice",
  "verificationMethod": [
    {
      "controller": "did:web:example.com:user:alice",
      "id": "did:web:example.com:user:alice#348ff0f0d0914251a625dfeac31a7470",
      "publicKeyJwk": {
        "alg": "EdDSA",
        "crv": "Ed25519",
        "kid": "348ff0f0d0914251a625dfeac31a7470",
        "kty": "OKP",
        "use": "sig",
        "x": "3XumG64Oz3oBr7kx7GOJvd7GO24wi28lXtn00hQs4I0"
      },
      "type": "Ed25519VerificationKey2019"
    }
  ]
}
```

**4. Now upload the DID document to your server, so that is can be resolved on the correct path**

In this example this should be:
`https://example.com/user/alice/did.json`
